spec-tests/src/test/java/org/candlepin/spec/ResponseFilteringSpecTest.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.candlepin.dto.api.client.v1.ConsumerDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.client.request.Request;
+import org.candlepin.spec.bootstrap.client.request.Response;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+
+@SpecTest
+class ResponseFilteringSpecTest {
+
+    private static ApiClient admin;
+    private static OwnerDTO owner;
+    private static ConsumerDTO consumer;
+
+    @BeforeAll
+    static void beforeAll() {
+        admin = ApiClients.admin();
+        owner = admin.owners().createOwner(Owners.random());
+        UserUtil.createUser(admin, owner);
+        consumer = admin.consumers().createConsumer(Consumers.random(owner));
+    }
+
+    @Test
+    void shouldAllowSingleExclude() {
+        ConsumerDTO consumerFull = admin.consumers().getConsumer(consumer.getUuid());
+        ConsumerDTO consumerFiltered = Request.from(admin)
+            .addQueryParam("exclude", "href")
+            .setPath("/consumers/" + consumer.getUuid())
+            .execute()
+            .deserialize(ConsumerDTO.class);
+
+        assertThat(consumerFull)
+            .usingRecursiveComparison()
+            .ignoringFields("href")
+            .isEqualTo(consumerFiltered);
+        assertThat(consumerFiltered.getHref()).isNull();
```

Comment:
Instead of null, can we check that the field is actually missing from the response?
I know that the href should never be null for a consumer, but in case that happened through a bug, this test would still pass for the wrong reason.

Same issue exists for all the other tests that assert on excluded fields

Reply:
Done.